### PR TITLE
Remove flags for watch disabling and source mode from EnsimeConfig

### LINKED ISSUE #1617: remove watch disabling from the .ensime
It is inconsistent to have server features mixed up with project configuration. Same goes for the "source mode" setting. This should be enabled by system properties for now and #1346 later.

### PR DESCRIPTION
**Pull request by yyadavalli**
Use system properties instead of `.ensime` flags

I'm not sure regarding the unit test, I don't think it is of any use now

Close #1617

### CODE REVIEW

**Comment by fommil** [comment]
I'll need to look at this one in some detail because my description of it in the ticket might not have been top quality. We still want to watch sources for when they are created or deleted, to update the source resolver. But we no longer want to watch sources to cause re-typechecking or updates to the source resolver unless the existence of the files has changed.

**Comment by yyadavalli** [comment]
I thought all you wanted to do is move the flags to disable watching from `EnsimeConfig`  to system properties (or environment variables), so that later they can be part of a `ServerConfig`

**Comment by fommil** [comment]
If that's all you're doing here then that's great. I wanted to remove the unnecessary work too, so maybe create another ticket.

File: core/src/test/scala/org/ensime/config/EnsimeConfigSpec.scala [hunk 1]
```diff
@@ -79,33 +78,28 @@ class EnsimeConfigSpec extends EnsimeSpec {
...
                 :targets ("$abc"))))""", { implicit config =>
-          config.sourceMode shouldBe sourceMode
-          config.compileClasspath shouldBe (
-            if (sourceMode) Set.empty else Set(abc)
-          )
-        })
-      }
-    }
+      config.compileClasspath shouldBe (
+        if (Properties.propOrFalse("ensime.sourceMode")) Set.empty else Set(abc)
```
Comment (yyadavalli) [inline core/src/test/scala/org/ensime/config/EnsimeConfigSpec.scala]: In that case, there is nothing this test would cover that's not already covered and can be deleted.

Comment (fommil) [inline core/src/test/scala/org/ensime/config/EnsimeConfigSpec.scala]: :+1:

**Comment by fommil** [comment]
weird, the `Test interactive search` emacs test failed. I can't see why it should be impacted :confused:

**Comment by yyadavalli** [comment]
Hmm, weird, I just checked and I have `Test inspector ui.` failing

**Comment by fommil** [comment]
That's probably because you're using Java 8.

**Pull request merged**